### PR TITLE
feat: AuthServerConnector/Provider

### PR DESCRIPTION
**Pull request by fzavalia**
Intended to connect the desktop client with the auth dapp.

### CODE REVIEW

File: src/providers/AuthServerProvider.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  private static authDappUrl: string = ''
+  private static identityExpirationInMillis: number = 30 * 24 * 60 * 60 * 1000 // 30 days in the future.
+
+  private chainId = ChainId.ETHEREUM_MAINNET
+  private account?: string
+
+  /**
+   * Set the url of the auth server to be consumed by this provider.
+   */
+  static setAuthServerUrl(url: string) {
```
Comment (LautaroPetaccio) [inline src/providers/AuthServerProvider.ts:20]: These method and variables will change every instance of the `AuthServerProvider` as they're static. Is this the intended behavior?

Comment (fzavalia) [reply src/providers/AuthServerProvider.ts:20]: Yes. So they can be configured once for all

File: src/connectors/AuthServerConnector.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    this.provider = new AuthServerProvider()
+  }
+
+  activate = async (): Promise<ConnectorUpdate<string | number>> => {
+    const previousAddress = localStorage.getItem(
+      AuthServerConnector.PREVIOUS_ADDRESS_KEY
+    )
+
+    if (previousAddress) {
+      const identity = sso.localStorageGetIdentity(previousAddress)
```
[the remark this answers is not in the document]
Comment (fzavalia) [reply src/connectors/AuthServerConnector.ts]: Moved it, looks much better 😄

File: src/connectors/AuthServerConnector.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  getChainId = async (): Promise<string | number> => {
+    return this.provider.getChainId()
+  }
+
+  getAccount = async (): Promise<string | null> => {
+    return this.provider.getAccount() ?? null
+  }
+
+  deactivate(): void {
+    localStorage.removeItem(AuthServerConnector.PREVIOUS_ADDRESS_KEY)
```
[the remark this answers is not in the document]
Comment (fzavalia) [reply src/connectors/AuthServerConnector.ts]: Yes!

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**